support-frontend/assets/helpers/redux/checkout/address/reducer.ts

```diff
@@ -68,6 +68,12 @@ function getAddressFieldsSlice(type: AddressType) {
 			setState(state, action: PayloadAction<string>) {
 				state.state = action.payload;
 				state.errors = removeError('state', state.errors);
+				const validationResult = addressFieldsSchema.safeParse(state);
+				if (!validationResult.success) {
+					state.errorObject = getSliceErrorsFromZodResult(
+						validationResult.error.format(),
+					);
+				}
```

Comment:
Rather than re-validating every time the state is updated, we should just delete any errors from the last Zod validation pass, as the validation will run again when the user next tries to submit the form.

```suggestion
				delete state.errorObject?.state;
```

Reply:
Have updated the code as per suggestion